### PR TITLE
[MAINTENANCE] Remove `dgtest` from dev pipeline

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- EOL `dgtest` dependency graph test runner in favor of `pytest`
- This marks the full removal of `dgtest` as a test strategy - moving forward, we will fully rely on `pytest` and `unit/integration/e2e` marking strategies

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: azure-pipelines-dev.yml [hunk 1]
```diff
@@ -326,41 +320,13 @@ stages:
...
-
-            displayName: 'dgtest-general-overrides'
-
-          - script: |
-              # These are tests that are specific to Great Expectations Cloud.
-              # In order to ensure coverage, we run them during each CI/CD cycle.
-              pytest tests --cloud -m "cloud and not e2e"
-
-            displayName: 'dgtest-cloud-overrides'
+            displayName: 'pytest'
```
Comment (anthonyburdi) [inline azure-pipelines-dev.yml:330]: Should we still run these "cloud and not e2e" tests? It looks like only "cloud AND e2e" tests were moved to a separate pipeline recently by @billdirks.

Comment (anthonyburdi) [reply azure-pipelines-dev.yml:330]: It looks like we are running the unit test cloud tests in a step above, should we also be running tests marked cloud + integration?

**Review by anthonyburdi — COMMENTED** [review]
One question about --cloud tests but otherwise looks good!

File: azure-pipelines-dev.yml (hunk 1 above)
Comment (cdkini) [reply azure-pipelines-dev.yml:330]: you are correct! added back this stage

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Comment by anthonyburdi** [comment]
Force merged since it is failing on the run of dgtest which is being removed.

**Pull request merged**